compiler/rustc_errors/src/lib.rs

```diff
@@ -1331,6 +1333,10 @@ impl DiagCtxtInner {
                     return if let Some(guar) = self.has_errors_or_lint_errors() {
                         Some(guar)
                     } else {
+                        // Is saving the diagnostic in `delayed_bugs` a notable
+                        // side-effect? Should `TRACK_DIAGNOSTIC` be called?
+                        // Unclear. Currently we err on the side of "no" to avoid
+                        // having to clone the diagnostic.
```

Comment:
We do not need to call `TRACK_DIAGNOSTIC`: if there is a delayed bug, the incremental session is deleted.

Reply:
Ok, I have updated the comment.

Reply:
:+1: